pluginlib/include/pluginlib/class_loader_imp.hpp

```diff
@@ -390,39 +392,77 @@ std::vector<std::string> ClassLoader<T>::getAllLibraryPathsToTry(
 
   const std::string path_separator = getPathSeparator();
 
-  std::vector<std::string> all_paths;
+  std::vector<std::string> all_paths;  // result of all pairs to search
+
   std::string package_prefix = ament_index_cpp::get_package_prefix(exporting_package_name);
-  std::vector<std::string> all_paths_without_extension = {
+
+  // Setup the directories to look in.
+  std::vector<std::string> all_search_paths = {
     // for now just try lib and lib64 (and their respective "libexec" directories)
     package_prefix + path_separator + "lib",
     package_prefix + path_separator + "lib64",
+    package_prefix + path_separator + "bin",  // also look in bin, for dll's on Windows
     package_prefix + path_separator + "lib" + path_separator + exporting_package_name,
     package_prefix + path_separator + "lib64" + path_separator + exporting_package_name,
+    package_prefix + path_separator + "bin" + path_separator + exporting_package_name,
   };
+
+  // Prepare to setup the relative file paths.
   bool debug_library_suffix = (0 == class_loader::systemLibrarySuffix().compare(0, 1, "d"));
```

Comment:
Note that currently `systemLibrarySuffix` will [always return the non debug version](https://github.com/ros/class_loader/commit/68796036d722808817b1ad54f7cb6600aa07107d) (but that's great to keep logic for handling both)